### ISSUE TITLE
Stackrox 3.73.1 introduced a breaking change for the Central Database component

### ISSUE THREAD

**Issue by balpert89** [stackrox/stackrox#4360]
Hi,

version 3.73.1 has introduced a breaking change where an older postgres version has been packaged:

3.72.1
```
sh-4.4$ postgres -V
postgres (PostgreSQL) 14.5
```

3.73.1
```
sh-4.4$ postgres -V
postgres (PostgreSQL) 13.9
```

This causes the central-db container to exit with a fatal error:
```
PostgreSQL Database directory appears to contain a database; Skipping initialization
2023-01-11 14:07:59.742 UTC [1] FATAL:  database files are incompatible with server
2023-01-11 14:07:59.742 UTC [1] DETAIL:  The data directory was initialized by PostgreSQL version 14, which is not compatible with this version 13.9.
```

**Comment by johannes94** [comment]
Hi @balpert89 , thanks for opening this issue. I'd like to ask you for some more context.

- In which shell/container did you run the command above?
- How did you deploy and upgrade stackrox?

**Comment by balpert89** [comment]
Hi @johannes94 

> In which shell/container did you run the command above?

I have run the commands in a container with the upstream images:
```
podman pull quay.io/stackrox-io/central-db:3.72.1
podman run -it quay.io/stackrox-io/central-db:3.72.1 /bin/sh
# in the running container
sh-4.4$ postgres -V
postgres (PostgreSQL) 14.5
sh-4.4$ exit

podman pull quay.io/stackrox-io/central-db:3.73.1
podman run -it quay.io/stackrox-io/central-db:3.73.1 /bin/sh
# in the running container
sh-4.4$ postgres -V
postgres (PostgreSQL) 14.5
sh-4.4$ exit
  ```

> How did you deploy and upgrade stackrox?
I have used the official helm charts (https://github.com/stackrox/helm-charts/tree/main/3.72.1 and https://github.com/stackrox/helm-charts/tree/main/3.73.1) and the manual in the README (https://github.com/stackrox/stackrox#manual-installation-using-helm).

**Comment by johannes94** [comment]
I've had a conversation on this in the dev team. It looks like we indeed downgraded the PostgreSQL version used going from `3.72.1` to `3.73.0`. It was a conscious decision we made due to compatibility to internal systems.

At the state of `3.72.1` the PostgreSQL integration was in active development and it only went to Technology Preview Phase when we released `3.73.0`. We can not guarantee that we don't introduce breaking changes in this phase of a features development. This will change once the PostgreSQL implementation is considered as stable (which is planned for `3.74.0`)

I know this explanation is not going to fix your issue here. If you can afford to loose the data, you can wipe out the data directory to allow PostgreSQL to reinitialize with 13.9. If not you can try to downgrade to the version that worked for you, create a database backup then reinitialize the PostgreSQL db using 13.9 and try importing backup. But I can't guarantee that this works.

**Comment by johannes94** [comment]
There is all kind of data stored in the central-db, basically everything the service needs to provide security analysis of a cluster. For example infos about policies, images, k8s roles, result of image scans.... .

If you're interested I would recommend inspecting the scheme e.g. by port-forwarding and connecting to the DB in your local environment. Also if you search for usage of the `RegisterTable` function in the code base you can find what tables are created.

I'll close this ticket with this comment, since the opening question was addressed. Thanks again for bringing up this issue and feel free to create other issues if something comes up :)
